weaviate/collection/classes/config.py

```diff
@@ -243,8 +388,8 @@ class CollectionConfigCreateBase(ConfigCreateModel):
     shardingConfig: Optional[ShardingConfigCreate] = Field(None, alias="sharding_config")
     vectorIndexConfig: Optional[VectorIndexConfigCreate] = Field(None, alias="vector_index_config")
     vectorIndexType: VectorIndexType = Field(VectorIndexType.HNSW, alias="vector_index_type")
-    vectorizer: Vectorizer = Vectorizer.NONE
-    moduleConfig: Dict[str, Any] = Field(None, alias="module_config")
+    vectorizer: Vectorizer = Field(Vectorizer.NONE)
+    moduleConfig: Optional[ModuleConfig] = Field(None, alias="module_config")
```

Comment:
I think we should combine these two options. I think it is a source of confusion/user-errors that you can set the vectorizer to MODULE1 and the moduleconfig for MODULE2.

How about we ditch `vectorizer` and rename `moduleConfig` to `vectorizer_options`? And then generate both entries in the dict?

Reply:
Are there instances where a user might want to supply their own custom-defined vectorizer that requires a string name that is unknown to us at this stage?

Reply:
yeah, probably. But then we should just offer a "Custom" option